lib/rubocop/cop/rspec/capybara/negation_matcher.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Enforces use of `have_no_*` or `not_to` for negated expectations.
+        #
+        # @example EnforcedStyle: have_no (default)
+        #   # bad
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        #   # good
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        # @example EnforcedStyle: not_to
+        #   # bad
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        #   # good
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        class NegationMatcher < Base
+          extend AutoCorrector
+          include ConfigurableEnforcedStyle
+
+          MSG = 'Use `%<good>s` instead of `%<bad>s`.'
+          CAPYBARA_MATCHERS = %w[
+            selector css xpath text title current_path link button
+            field checked_field unchecked_field select table
+            sibling ancestor
+          ].freeze
+          POSITIVE_MATCHERS = CAPYBARA_MATCHERS.flat_map do |element|
+            ["have_#{element}".to_sym]
+          end.freeze
```

Comment:
Why using flat_map with single elements?
```suggestion
          POSITIVE_MATCHERS = CAPYBARA_MATCHERS.map do |element|
            "have_#{element}".to_sym
          end.freeze
```

Reply:
Oops, you're right. I fixed it.